### PR TITLE
Round PopupMenu ripple corners

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/1200204095367872/1201672589972569/f

### Description

- Ensures that the first item's ripple has rounded corners on the top left and top right.

- Ensures that the last item's ripple has rounded corners on the bottom left and bottom right.

- If there is only one item in the `PopupMenu`, all of the ripple corners are rounded.

### Steps to test this PR

- [x] Add a favorite
- [x] Long press the favorite on a new tab
- [x] Long press the items in the `PopupMenu`
- [x] Verify that the ripple does not extend past the corners of the `PopupMenu`

### UI changes
| Before  | After |
| ------ | ----- |
Screenshot_20240624_100616|Screenshot_20240624_100715

### CODE REVIEW

**Comment by joshliebe** [comment]
* **#4681** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4681?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joshliebe and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: common/common-ui/src/main/res/drawable/ripple_bottom_rounded.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+    android:color="?colorControlHighlight">
+    <item android:id="@android:id/mask">
+        <shape android:shape="rectangle">
+            <solid android:color="@color/black" />
+            <corners android:bottomLeftRadius="8dp" android:bottomRightRadius="8dp" />
```
Comment (malmstein) [inline common/common-ui/src/main/res/drawable/ripple_bottom_rounded.xml:22]:
```suggestion
            <corners android:bottomLeftRadius=“@dimen/smallShapeCornerRadius" android:bottomRightRadius="@dimen/smallShapeCornerRadius" />
```

File: common/common-ui/src/main/res/drawable/ripple_top_rounded.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+    android:color="?colorControlHighlight">
+    <item android:id="@android:id/mask">
+        <shape android:shape="rectangle">
+            <solid android:color="@color/black" />
+            <corners android:topLeftRadius="8dp" android:topRightRadius="8dp" />
```
Comment (malmstein) [inline common/common-ui/src/main/res/drawable/ripple_top_rounded.xml:22]:
```suggestion
            <corners android:topLeftRadius=""@dimen/smallShapeCornerRadius" android:topRightRadius=""@dimen/smallShapeCornerRadius" />
```

**Review by malmstein — APPROVED** [review]
nice one @joshliebe! I’ve added a suggestion to use dimensions, but it’s looking good.

**Pull request merged**